src/plugins/validation.ts

```diff
@@ -0,0 +1,388 @@
+    }
+    return this._summary.status === 'success';
+  }
+
+  /**
+   * Transform the report to a well formatted table string.
+   */
+  public toString(): string {
+
+    const json = this.toJson();
+    const output = [json.title];
+
+    output.push('-'.repeat(json.title.length));
+    output.push('');
+    output.push('(Summary)');
+    output.push('');
+    output.push(table([
+      ['Status', json.summary.status],
+      ['Plugin', json.summary.plugin],
+      ['Version', json.summary.version],
+      ...Object.entries(json.summary.metadata ?? {}),
+    ]));
+
+    if (json.violations) {
+      output.push('');
+      output.push('(Violations)');
+    }
+    for (const violation of json.violations) {
+      const occurrences = violation.violatingConstructs.flatMap(c => c.locations).length;
+      const title = `\x1b[1m\x1b[31m${violation.ruleName} (${occurrences} occurrences)\x1b[0m`;
+      output.push('');
+      output.push(title);
+      output.push('');
+      output.push('  Occurrences:');
+      for (const construct of violation.violatingConstructs) {
+        output.push('');
+        output.push(`    - construct.path: ${construct.constructPath ?? 'N/A'}`);
+        output.push(`    - manifest.path: ${construct.manifestPath}`);
+        output.push(`    - resource.name: ${construct.resourceName}`);
+        if (construct.locations) {
+          output.push('    - locations:');
+          for (const location of construct.locations) {
+            output.push(`      > ${location}`);
+          }
+        }
+      }
+      output.push('');
+      output.push(`  Recommendation: ${violation.recommendation}`);
+      output.push(`  How to fix: ${violation.fix}`);
+    }
+
+    return output.join(os.EOL);
+
+  }
+
+  /**
+   * Transform the report into a JSON object.
+   */
+  public toJson(): ValidationReportJson {
+    if (!this._summary) {
+      throw new Error('Unable to determine report result: Report is incomplete. Call \'report.submit\'');
+    }
+    return {
+      title: `Validation Report (${this.pkg}@${this.version})`,
+      violations: this.violations,
+      summary: this._summary,
+    };
+  }
+
+}
+
+/**
+ * Utiliy class for loading validation plugins.
+ */
+export class ValidationPlugin {
+
+  /**
+   * Load the validation plugin and create the necessary context for its execution.
+   */
+  public static load(
+    validation: ValidationConfig,
+    manifests: string[],
+    stdout: boolean,
+    pluginManager: PluginManager): { plugin: Validation; context: ValidationContext } {
+
+    const plugin = pluginManager.load({
+      pkg: validation.package,
+      version: validation.version,
+      class: validation.class,
+      properties: validation.properties,
+      installEnv: validation.installEnv,
+    });
+
+    if (typeof plugin.instance.validate !== 'function') {
+      throw new Error(`Instance of class ${validation.class} is not a validation plugin. Are you sure you specified the correct class?`);
+    }
```

Comment:
Might be useful to specify the package name & version, too... there may be multiple plug-ins with the same class name configured...

Reply:
Done